components/brave_wallet/browser/fil_transaction_unittest.cc

```diff
@@ -30,119 +41,373 @@ TEST(FilTransactionUnitTest, Initialization) {
 }
 
 TEST(FilTransactionUnitTest, Serialization) {
-  auto transaction = FilTransaction::FromTxData(
-      mojom::FilTxData::New("1", "2", "3", "4", "5",
-                            "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q", "6"));
+  auto transaction = FilTransaction::FromTxData(mojom::FilTxData::New(
+      "1", "2", "3", "4", "5", "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q",
+      "t1h5tg3bhp5r56uzgjae2373znti6ygq4agkx4hzq", "6"));
   EXPECT_EQ(transaction, FilTransaction::FromValue(transaction->ToValue()));
 
-  auto empty_nonce = FilTransaction::FromTxData(
-      mojom::FilTxData::New("", "2", "3", "1", "5",
-                            "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q", "6"));
+  auto empty_nonce = FilTransaction::FromTxData(mojom::FilTxData::New(
+      "", "2", "3", "1", "5", "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q",
+      "t1h5tg3bhp5r56uzgjae2373znti6ygq4agkx4hzq", "6"));
   EXPECT_EQ(empty_nonce, FilTransaction::FromValue(empty_nonce->ToValue()));
 }
 
-TEST(FilTransactionUnitTest, GetMessageToSign) {
-  auto transaction = FilTransaction::FromTxData(
-      mojom::FilTxData::New("1", "2", "3", "1", "5",
-                            "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q", "6"));
-  EXPECT_EQ(transaction->GetMessageToSign(),
-            "{\"gas_fee_cap\":\"3\",\"gas_limit\":\"1\",\"gas_"
-            "premium\":\"2\",\"max_fee\":\"5\",\"nonce\":\"1\",\"to\":"
-            "\"t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q\",\"value\":\"6\"}");
+TEST(FilTransactionUnitTest, GetMessageToSignSecp) {
+  auto transaction = FilTransaction::FromTxData(mojom::FilTxData::New(
+      "1", "2", "3", "1", "5", "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q",
+      "t1h5tg3bhp5r56uzgjae2373znti6ygq4agkx4hzq", "6"));
+  auto message_to_sign = transaction->GetMessageToSign();
+  ASSERT_TRUE(message_to_sign);
+  CompareJSONs(*message_to_sign,
+               R"({
+                 "From": "t1h5tg3bhp5r56uzgjae2373znti6ygq4agkx4hzq",
+                 "GasFeeCap": "3",
+                 "GasLimit": 1,
+                 "GasPremium": "2",
+                 "MethodNum": 0,
+                 "Params": "",
+                 "Nonce": 1,
+                 "To": "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q",
+                 "Value": "6",
+                 "Version": 0
+               })");
+
+  auto signature = transaction->GetSignedTransaction(
+      "8VcW07ADswS4BV2cxi5rnIadVsyTDDhY1NfDH19T8Uo=");
+  ASSERT_TRUE(signature.has_value());
+  auto signature_value = base::JSONReader::Read(*signature);
+  EXPECT_TRUE(signature_value);
+  auto* message = signature_value->FindKey("Message");
+  auto* signature_data = signature_value->FindStringPath("Signature.Data");
+  EXPECT_TRUE(message);
+  EXPECT_TRUE(signature_data);
+  auto message_as_value = base::JSONReader::Read(*message_to_sign);
+  EXPECT_TRUE(message_as_value);
+  EXPECT_EQ(*signature_data,
+            "SozNIZGNAvALCWtc38OUhO9wdFl82qESGhjnVVhI6CYNN0gP5qa+hZtyFh+"
```

Comment:
Are these hard coded values generated by our signature implementation for this test or are they standard test vectors that we got from somewhere?

Reply:
genereated by this test

Reply:
Let's wait and see on what others thoughts are above for the underlying crypto libs. I'm going to resolve this comment for now to consolidate the conversation to the thread above